### ISSUE TITLE
at_onboarding_flutter broken in version 4.0.4 crashes app !

### ISSUE THREAD

**Issue by cconstab** [atsign-foundation/at_widgets#469]
### Which at_widget package you find issue with?

at_onboarding_flutter

### On what platforms did you experience the issue?

Windows, MacOS

### What version are you using?

package version 4.0.4 (this is due to at_app not allowing 5.0.0)

### What happened?

Onboarding widget no longer works at all and crashes on macos also secret keys an print statements are logged on the console which should NEVER happen..

This was working just yesterday !

```
flutter: File: 'G:\My Drive\@keys\@visual61_key.atKeys'    <<<<<<<<<<<<<<<<
flutter: ATSIGN>>>visual61
flutter: AES>>>>>vhhfjksdhfjksdf********************=
```

My key is truncated with `*` but you get the picture!!!

Windows does not crash but does not work macos crashed and red
screens the app..

```
cconstab@tarial nautel_app % flutter run -d macos
Running "flutter pub get" in nautel_app...                       2,232ms
Launching lib/main.dart on macOS in debug mode...
Running pod install...                                             835ms
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00006001-000C59191441801E }
{ platform:macOS, arch:x86_64, id:00006001-000C59191441801E }
Building macOS application...                                           
Syncing files to device macOS...                                    92ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on macOS is available at: http://127.0.0.1:52173/Hsm4mIHrF0A=/
The Flutter DevTools debugger and profiler on macOS is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:52173/Hsm4mIHrF0A=/
flutter: INFO|2022-08-11 17:43:02.840344|HiveBase|commit_log_678498516f1217944ff86bc8dbf9a680b8bb87c183b3f4ef867ad2f77ef50ac2 initialized successfully 

flutter: SEVERE|2022-08-11 17:43:02.855281|HivePersistenceManager|AtPersistence.init exception: FileSystemException: lock failed, path = '/Users/cconstab/Library/Containers/com.example.nautelApp/Data/Library/Application Support/com.example.nautelApp/678498516f1217944ff86bc8dbf9a680b8bb87c183b3f4ef867ad2f77ef50ac2.lock' (OS Error: Resource temporarily unavailable, errno = 35) 

[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: FileSystemException: lock failed, path = '/Users/cconstab/Library/Containers/com.example.nautelApp/Data/Library/Application Support/com.example.nautelApp/678498516f1217944ff86bc8dbf9a680b8bb87c183b3f4ef867ad2f77ef50ac2.lock' (OS Error: Resource temporarily unavailable, errno = 35)
#0      _RandomAccessFile.lock.<anonymous closure> (dart:io/file_impl.dart:1002:9)
#1      _rootRunUnary (dart:async/zone.dart:1434:47)
#2      _CustomZone.runUnary (dart:async/zone.dart:1335:19)
<asynchronous suspension>
#3      StorageBackendVm.initialize (package:hive/src/backend/vm/storage_backend_vm.dart:81:5)
<asynchronous suspension>
#4      HiveImpl._openBox (package:hive/src/hive_impl.dart:111:9)
<asynchronous suspension>
#5      HiveImpl.openLazyBox (package:hive/src/hive_impl.dart:160:12)
<asynchronous suspension>
#6      HiveBase.openBox (package:at_persistence_secondary_server/src/keystore/hive_base.dart:24:9)
<asynchronous suspension>
#7      HivePersistenceManager.initialize (package:at_persistence_secondary_server/src/keystore/hive_manager.dart:33:7)
<asynchronous suspension>
#8      HiveBase.init (package:at_persistence_secondary_server/src/keystore/hive_base.dart:15:5)
<asynchronous suspension>
#9      StorageManager._initStorage (package:at_client/src/manager/storage_manager.dart:42:5)
<asynchronous suspension>
#10     StorageManager.init (package:at_client/src/manager/storage_manager.dart:22:7)
<asynchronous suspension>
#11     AtClientImpl.create (package:at_client/src/client/at_client_impl.dart:110:7)
<asynchronous suspension>
#12     AtClientManager.setCurrentAtSign (package:at_client/src/manager/at_client_manager.dart:57:24)
<asynchronous suspension>
#13     AtClientService._init (package:at_client_mobile/src/at_client_service.dart:23:23)
<asynchronous suspension>
#14     AtClientService.onboard (package:at_client_mobile/src/at_client_service.dart:249:5)
<asynchronous suspension>
#15     OnboardingService.onboard (package:at_onboarding_flutter/services/onboarding_service.dart:83:19)
<asynchronous suspension>
#16     _FutureBuilderState._subscribe.<anonymous closure> (package:flutter/src/widgets/async.dart:627:33)
<asynchronous suspension>

flutter: INFO|2022-08-11 17:43:02.861337|At Onboarding Flutter|Onboarding...! 


══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following LateError was thrown building Builder:
LateInitializationError: Field 'notificationService' has not been initialized.

The relevant error-causing widget was:
  MaterialApp
  MaterialApp:file:///Users/cconstab/Documents/GitHub/cconstab/at_nautel_snmp/nautel_app/lib/screens/home_screen.dart:26:12

When the exception was thrown, this was the stack:
#0      AtClientManager.notificationService (package:at_client/src/manager/at_client_manager.dart)
#1      _MyHomePageState.initState (package:nautel_app/screens/home_screen.dart:61:63)
#2      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4942:57)
#3      ComponentElement.mount (package:flutter/src/widgets/framework.dart:4781:5)
...     Normal element mounting (24 frames)
#27     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3817:16)
#28     MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:6350:36)
#29     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6362:32)
...     Normal element mounting (175 frames)
#204    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3817:16)
#205    MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:6350:36)
#206    MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6362:32)
...     Normal element mounting (385 frames)
#591    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3817:16)
#592    MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:6350:36)
#593    MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6362:32)
...     Normal element mounting (175 frames)
#768    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3817:16)
#769    MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:6350:36)
#770    Element.updateChild (package:flutter/src/widgets/framework.dart:3551:18)
#771    RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5883:32)
#772    MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6375:17)
#773    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#774    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#775    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#776    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#777    StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#778    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#779    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#780    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#781    ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#782    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#783    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#784    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#785    ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#786    _InheritedNotifierElement.update (package:flutter/src/widgets/inherited_notifier.dart:108:11)
#787    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#788    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6222:14)
#789    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#790    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#791    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#792    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#793    StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#794    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#795    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6222:14)
#796    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#797    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6222:14)
#798    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#799    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#800    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#801    ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#802    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#803    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#804    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#805    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#806    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2659:19)
#807    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:891:21)
#808    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:370:5)
#809    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1146:15)
#810    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1083:9)
#811    SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:997:5)
#815    _invoke (dart:ui/hooks.dart:151:10)
#816    PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#817    _drawFrame (dart:ui/hooks.dart:115:31)
(elided 3 frames from dart:async)

════════════════════════════════════════════════════════════════════════════════════════════════════
```

### Steps to reproduce

```bash
1. Create a flutter app with at_app
2. Run 
3. Authenticate an atSign using known keys
4. see the crash
...
```

### Flutter analyze output

```console
cconstab@tarial nautel_app % flutter analyze
Analyzing nautel_app...                                                 

   info • Unused import: 'package:at_onboarding_flutter/at_onboarding_flutter.dart' • lib/main.dart:6:8 • unused_import
   info • Prefer declaring const constructors on `@immutable` classes • lib/screens/home_screen.dart:44:3 •
          prefer_const_constructors_in_immutables
   info • Prefer const with constant constructors • lib/screens/home_screen.dart:169:24 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/screens/home_screen.dart:187:24 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/screens/home_screen.dart:205:24 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/screens/home_screen.dart:223:24 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/screens/home_screen.dart:241:24 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/screens/home_screen.dart:259:24 • prefer_const_constructors
   info • Unused import: 'package:intl/intl.dart' • lib/transmitter_lookup.dart:2:8 • unused_import
   info • Unused import: 'package:http/http.dart' • lib/transmitter_lookup.dart:4:8 • unused_import
   info • Name source files using `lowercase_with_underscores` • lib/widgets/Gaugewidget.dart:1:1 • file_names
   info • Prefer declaring const constructors on `@immutable` classes • lib/widgets/Gaugewidget.dart:24:3 •
          prefer_const_constructors_in_immutables
   info • Use key in widget constructors • lib/widgets/Gaugewidget.dart:24:3 • use_key_in_widget_constructors
   info • Avoid `print` calls in production code • lib/widgets/Gaugewidget.dart:49:5 • avoid_print
   info • Unnecessary string interpolation • lib/widgets/Gaugewidget.dart:81:20 • unnecessary_string_interpolations
   info • Prefer const with constant constructors • lib/widgets/Gaugewidget.dart:81:52 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/widgets/Gaugewidget.dart:86:22 • prefer_const_constructors
   info • Unnecessary string interpolation • lib/widgets/Gaugewidget.dart:88:15 • unnecessary_string_interpolations
   info • Prefer const with constant constructors • lib/widgets/Gaugewidget.dart:89:22 • prefer_const_constructors
   info • Prefer typing uninitialized variables and fields • lib/widgets/Gaugewidget.dart:99:9 • prefer_typing_uninitialized_variables
   info • Prefer typing uninitialized variables and fields • lib/widgets/Gaugewidget.dart:127:9 • prefer_typing_uninitialized_variables
   info • Prefer typing uninitialized variables and fields • lib/widgets/Gaugewidget.dart:155:9 • prefer_typing_uninitialized_variables
   info • The value of the local variable 'result' isn't used • lib/widgets/Gaugewidget.dart:155:9 • unused_local_variable
   info • Avoid empty statements • lib/widgets/Gaugewidget.dart:162:9 • empty_statements
   info • Avoid empty statements • lib/widgets/Gaugewidget.dart:166:9 • empty_statements
   info • Avoid empty statements • lib/widgets/Gaugewidget.dart:170:9 • empty_statements
   info • Avoid empty statements • lib/widgets/Gaugewidget.dart:174:9 • empty_statements
   info • Avoid empty statements • lib/widgets/Gaugewidget.dart:178:9 • empty_statements
   info • The operand can't be null, so the condition is always true • lib/widgets/onboarding_dialog.dart:40:25 •
          unnecessary_null_comparison
   info • Prefer const with constant constructors • lib/widgets/onboarding_dialog.dart:266:18 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/widgets/onboarding_dialog.dart:270:22 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/widgets/onboarding_dialog.dart:279:22 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/widgets/onboarding_dialog.dart:279:43 • prefer_const_constructors

33 issues found. (ran in 1.5s)
cconstab@tarial nautel_app %
```

### Flutter doctor output

```console
cconstab@tarial nautel_app % flutter doctor -v
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.5 21G72 darwin-arm, locale en-US)
    • Flutter version 3.0.5 at /Users/cconstab/flutter-arm64
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (4 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/cconstab/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.70.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (3 available)
    • iPhone13 (mobile) • C82AB7A2-AFC2-4EE0-ACBE-AB32F0C6A7CF • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)   • macos                                • darwin-arm64   • macOS 12.5 21G72 darwin-arm
    • Chrome (web)      • chrome                               • web-javascript • Google Chrome 104.0.5112.79

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
cconstab@tarial nautel_app %
```

**Comment by cconstab** [comment]
So looking through the error messages it looked like this is NOT a at_onboarding_flutter issue directly, but is something we need to fix.

 In the background i had another instance of the app running both on my Windows machine and Mac. When onboarding trys to open the hive boxes it fails (see the logs above) But then wierd stuff happens, the app crashes on mac but does od stuff on Windows.

 We I think either need to allow multiple instances to run in sandboxes like android/ios do so each app has its own hive box or we need to fail gracefully when we fail to open the boxes.
 
 
 However the printing of the AES keys needs to be fixed ASAP!!

**Comment by cconstab** [comment]
Windows still not working 🤔

**Comment by VJag** [comment]
@murali-shris @sachins-geekyants any idea where the printing of AES keys is happening?

@murali-shris @kalluriramkumar  I think we need to call close() on Hive boxes on shutdown hook equivalents in Dart as a best practice.

**Comment by cconstab** [comment]
https://user-images.githubusercontent.com/6131216/184284023-95f14ca6-b9a7-4934-afce-eaee30b3c8e5.mp4

**Comment by cconstab** [comment]
> @murali-shris @sachins-geekyants any idea where the printing of AES keys is happening?
> 
> @murali-shris @kalluriramkumar I think we need to call close() on Hive boxes on shutdown hook equivalents in Dart as a best practice.


at_onboarding_flutter-4.0.4\lib\screens\pair_atsign.dart
>

**Comment by VJag** [comment]
@cconstab So are we running the same app twice?

**Comment by VJag** [comment]
> video1715746103.mp4

@sachins-geekyants Can you look into this?

**Comment by cconstab** [comment]
> @cconstab So are we running the same app twice?
> 
> 

On Mac I was hence the logs above..

On windows the same code produces a broken experience as per video

**Comment by VJag** [comment]
Thinking out loud, @murali-shris @kalluriramkumar  Does our give box file path naming convention friendly to run the same app twice? 🤔

**Comment by sachins-geekyants** [comment]
@cconstab I was able to replicate the issue and also got the red screen in macOS.

Actually, onboarding process was failing and `_handleError`  is getting called.
If you replace onPressed function body

`Navigator.of(context).pushReplacementNamed(HomeScreen.id);` 

with 
``` 
 if (Navigator.of(context).canPop()) {
                Navigator.of(context).pop();
}
```
It will not move to next screen and no red screen would show up.

**Comment by sachins-geekyants** [comment]
Also, when running multiple instances of the app, race condition is happening and OS is not able to read hive directory.
Found a similar reported issue : https://github.com/hivedb/hive/issues/333

**Comment by cconstab** [comment]
@sachins-geekyants thanks for help.. On Windows this is the error I get when trying to onboard and hence the pop up of the "Unable to onboard widget" (although it has onboarded later)

I put in a print statement 

```
            if ((atsign != null) && !(_atSignsList.contains(atsign))) {
              setState(() {
                _atSignsList;
                _atSignsList.add(atsign);
                _atsign = atsign;
              });
            }
            Navigator.of(context).pushNamed(HomeScreen.id);
          },
          onError: (error) {
            print(error.toString());
            _handleError(context);
          }
```

And see this as the Error

```
flutter: OnboardingStatus.ENCRYPTION_PRIVATE_KEY_NOT_FOUND
```

Again this seems only to happen on Windows so suspect it is a at_onboarding_flutter issue specific to windows.. It may also explain the Print statements displaying the AEK key and atSign!

Could you take another look specific to Windows ?

**Comment by sachins-geekyants** [comment]
I tried running for windows but Amazon workspace does not allows to authenticate because of following certificate issue:
```
flutter: SEVERE|2022-08-16 18:14:03.627358|AtLookup|AtLookup.findSecondary connection to root.atsign.org exception: HandshakeException: Handshake error in client (OS Error:
        CERTIFICATE_VERIFY_FAILED: unable to get local issuer certificate(../../third_party/boringssl/src/ssl/handshake.cc:393))
```
Tried adding this suggested workaround in at_lookup, but didn't worked: https://stackoverflow.com/questions/54285172/how-to-solve-flutter-certificate-verify-failed-error-while-performing-a-post-req

I have requested for a windows system from geekyAnts to further test on it.


Also, **Onboarding 5.0.0** was made from scratch taking reference from **4.0.0**.
The issue which is coming in 4.0.0 might not come in 5.0.0. 

We should first try to migrate to 5.0.0 and if this issue still comes up, we can debug it.
Xavier has made a PR for at_app_flutter migration to use 5.0.0 : https://github.com/atsign-foundation/at_app/pull/175

**Comment by sachins-geekyants** [comment]
@cconstab Can you once try running with at_onboarding_flutter:4.0.3/4.0.2 ?

**Comment by cconstab** [comment]
I migrated (eventually to at_onboarding_flutter 5.0.0) and I get the same issue with the encryption keys 

See video

https://user-images.githubusercontent.com/6131216/185002817-828de2de-dd31-4a0e-8d1c-5bf4a0ab5f29.mp4

**Comment by cconstab** [comment]
@sachins-geekyants I see the keys being pickup in keychain_manager.dart but the keys are not getting into at_client_service.dart but only on Windows..

As I see them in keychain manager the Windows/Biometric storage is working..  So odd


 I have a 5.0.0. branch now that is sorta working on everything except Windows still..
 
 
 https://github.com/cconstab/at_nautel_snmp

**Comment by sachins-geekyants** [comment]
We are getting exception in saving `AtClientData` data in `BiometricStorageFile`.
When AtClientData just have `pkamPrivateKey` , `pkamPrivateKey` it saves the record.

But when AtClientData have `pkamPrivateKey` , `pkamPrivateKey`, `encryptionPublicKey`, `encryptionPrivateKey` it throws following exception.

```
[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: BiometricStorageException{message: Error writing credential design.codeux.authpass.default_unauthenticated (0): 0}
#0      Win32BiometricStoragePlugin.write (package:biometric_storage/src/biometric_storage_win32.dart:122:9)
#1      BiometricStorageFile.write (package:biometric_storage/src/biometric_storage.dart:441:15)
#2      StorageActions.build.<anonymous closure> (package:test_biometric/main.dart:275:20)
#3      StorageActions.build.<anonymous closure> (package:test_biometric/main.dart:271:22)
#4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:1005:21)
#5      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:198:24)
#6      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:613:11)
#7      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:298:5)
#8      BaseTapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:232:7)
#9      PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:563:9)
#10     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:94:12)
#11     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:139:9)
#12     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:614:13)
#13     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:137:18)
#14     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:123:7)
#15     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:445:19)
#16     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:425:22)
#17     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:329:11)
#18     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:380:7)
#19     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:344:5)
#20     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:302:7)
#21     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:285:7)
#22     _rootRunUnary (dart:async/zone.dart:1442:13)
#23     _CustomZone.runUnary (dart:async/zone.dart:1335:19)
#24     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
#25     _invoke1 (dart:ui/hooks.dart:170:10)
#26     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:331:7)
#27     _dispatchPointerDataPacket (dart:ui/hooks.dart:94:31)
```

Windows is limiting on data that can be saved BiometricStorageFile.

**Comment by sonle-geekyants** [comment]
The issues happened because `biometric_storage` package limits the number of characters saved in a key.
- In Android max is 2147483647 characters
- In iOS max is 4.2 billion characters
- But in Window max is 2560 characters. https://docs.microsoft.com/en-us/windows/win32/api/wincred/ns-wincred-credentiala
As everyone knows that with > `at_client_mobile 3.2.3+1` I did combine all data to one key before saved to biometric_storage.

So, please use override package: `at_client_mobile: 3.1.19` for hotfix in Window. I will fix this issue in the next `at_client_mobile`.

**Comment by gkc** [comment]
What are you thinking with regards to how to fix the issue, @sonle-geekyants ?

**Comment by sonle-geekyants** [comment]
@gkc I think that we can separate the data into many segments and save it in many keys for Window platform.
But I will find some other better solutions before deciding how to fix the issue.
If you have any ideas, please let me know. Thank you!

**Comment by sonle-geekyants** [comment]
@cconstab, @gkc , and @sachins-geekyants 
I have updated `at_client_mobile` for Window platform in this branch `hotfix/window-limit-data-length`
Please override the package and test it again.

This is my commit: https://github.com/atsign-foundation/at_client_sdk/commit/cf4866449307a17ccaf70510b5dfafd44d3a5892
And the change just impacts to window platform only.

**Comment by cconstab** [comment]
@sonle-geekyants Nice work!! This is now working on Windows! I think it would be worth documenting how we store the keys on each platform and why Windows is a little different on the atsign.dev website (v3.0) @anthonyvprakash  could you help on that ?

Thank you so much.. @gkc  make sure you like the way this is being solved but I am super happy you found the solution @sonle-geekyants 
@sachins-geekyants

**Comment by sachins-geekyants** [comment]
Key changes for windows have been merged into at_client_mobile trunk, waiting for a patch release so that onboarding package can be updated.

**Comment by sachins-geekyants** [comment]
Fix released with `at_onboarding_flutter 5.0.1`

**Comment by cconstab** [comment]
Wonderful! Nice work thanks
